packages/rs-drive-abci/src/execution/platform_events/identity_credit_withdrawal/update_broadcasted_withdrawal_transaction_statuses/v0/mod.rs

```diff
@@ -59,20 +63,39 @@ where
             )));
         };
 
-        let core_transactions = self.fetch_core_block_transactions(
```

Comment:
It seems that both approach has own benefits. When system under normal load (we have blocks one after another) the block reading approach is more efficient because we will make a call to take a block once 2,5 minutes. But if we check transactions we will have call every block with multiple operations in Core to check transactions.

The new approach has benefits when we don't have load at all (no or very few withdrawals and rare blocks) and chain halt.

Potentially we could make hybrid approach and see how much transactions / block to read we have and choose one or another.

Reply:
No the other method was never ever more efficient.

Remember transactions are indexed in core on nodes that run evolution, so when we ask for 5 transactions, we just use that index, and we do so in one call. The only situation where the old system might be faster is when there is one core block with less than 10 transactions, but it would be on the order of 0.005ms to 0.006ms, both very fast.